src/renderer/components/message/LabeledLink.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useContext, useState } from 'react'
+import { ScreenContext } from '../../contexts'
+import { toASCII } from 'punycode'
+import { OpenDialogFunctionType } from '../dialogs/DialogController'
+import { SmallDialog } from '../dialogs/DeltaDialog'
+import { Classes } from '@blueprintjs/core'
+import { DeltaCheckbox } from '../contact/ContactListItem'
+import { getLogger } from '../../../shared/logger'
+
+const log = getLogger('renderer/LabeledLink')
+
+const { openExternal } = window.electron_functions
+
+function getTrustedDomains(): string[] {
+  return JSON.parse(localStorage.getItem('trustedDomains') || '[]')
+}
+
+function trustDomain(domain: string) {
+  log.info('trustDomain', domain)
+  const trusted: string[] = getTrustedDomains()
+  trusted.push(domain)
+  localStorage.setItem('trustedDomains', JSON.stringify(trusted))
+}
+
+function isDomainTrusted(domain: string): boolean {
+  return getTrustedDomains().includes(domain)
+}
+
+export const LabeledLink = ({
+  label,
+  target,
+}: {
+  label: string
+  target: string
+}) => {
+  const { openDialog } = useContext(ScreenContext)
+  const splittedTarget = String(target).split('/')
+  // encode the punycode to make phishing harder
+  const domain = (splittedTarget[2] = splittedTarget[2]
+    .split('.')
+    .map(toASCII)
+    .join('.'))
+  const sanitizedTarget = splittedTarget.join('/')
+  const onClick = (ev: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    ev.preventDefault()
+    ev.stopPropagation()
+    //check if domain is trusted
+    if (isDomainTrusted(domain)) {
+      openExternal(target)
+      return
+    }
+    // not trusted - ask for confimation from user
+    ;(openDialog as OpenDialogFunctionType)(({ isOpen, onClose }) => {
```

Comment:
Can you factor this out into an own LinkDialogFunction? I'm fine with this living in this file.